languages/swift/exercises/concept/string-components/.docs/instructions.md

```diff
@@ -0,0 +1,151 @@
+
+The guard will then give you the word formed by the array of letters you replied with for you to put into capitalized word form and append `", please."`. Finally the password you return is `Horse, please.`, and you'll get in.
+
+## Secret room
+
+Inside the back room of the club is another door that leads to the secret room that only the very top poets may enter.
+
+1. The guard will recite a poem;
+   - Again, you will have to split the poem into lines and respond with the appropriate letter _but now the appropriate letter from line number i is the i<sup>th</sup> letter of the line_.
+2. The guard will tell you all the letters you've responded with at once:
+   - You need to format the letters as a capitalized word
+   - You must shout the secret phrase by returning an uppercased string with an exclamation point added to the end.
+
+For example, a modified version of the poem mentioned before fits this pattern:
+
+```text
+Stands so high
+Huge hooves too
+Impatiently waits for
+Rider with harness
+Eager to leave
+```
+
+When the guard recites the poem, you will split it into individual lines, strip off any trailing spaces, and respond with the i<sup>th</sup> letters of each line, i.e. ["S", "u", "p", "e", "r"].
+
+The guard will then give you the word formed by the array of letters you replied with for you to put into uppercased word form and append `"!"`. Finally the password you return is `SUPER!`, and you'll get in.
+
+## Tasks
+
+## 1. Split a string into individual lines
+
+Implement the function `splitLines(_:)` that takes a `String` as output and splits it into an array of `String`s using newlines as delimiters.
+
+```swift
+splitLines("Hello.\nHow are you?\n\nI'm doing fine.")
+// => ["Hello." ,"How are you?", "", "I'm doing fine"].
+```
+
+## 2. Get the first letter of a sentence
+
+Implement the function `firstLetter(_:)` that returns first letter of a sentence If there is no last letter, return an underscore (`_`):
+
+```swift
+firstLetter("Stands so high")
+// => "S"
+```
+
+## 3. Capitalize a word
+
+Implement the function `capitalize(_:)` that correctly capitalizes a word:
+
+```swift
+capitalize("SHIRE")
+// => "Shire"
+
+capitalize("horse")
+// => "Horse"
+```
+
+## 4. Trim a sentence
+
+Implement the function `trimFromEnd(_:)` that removes whitespace from the end of a sentence and returns the trimmed sentence:
+
+```swift
+trimFromEnd("Stands so high   ")
+// => "Stands so high"
+```
+
+## 5. Get the last letter of a sentence
+
+Implement the function `lastLetter(_:)` that returns the last letter of a sentence. If there is no last letter, return an underscore (`_`):
+
+```swift
+lastLetter("Stands so high")
+// => "h"
+```
+
+## 6. Be polite
+
+Implement the function `backDoorPassword(_:)` that takes a string as input and formats it in the polite manner required for the backdoor password:
+
+```swift
+backDoorPassword('horse')
```

Comment:
Both single- and double quotes are used to define strings. It might for the student to see one consistent format?

Reply:
Weird. No idea how that crept in there, must've been a copy-paste from the exercise I cloned the original from. Swift doesn't even do anything with single quotes at the moment.